types/middlewares.d.ts

```diff
@@ -5,6 +5,7 @@ declare module 'mongoose' {
   type MongooseDistinctDocumentMiddleware = 'save' | 'init' | 'validate';
   type MongooseDocumentMiddleware = MongooseDistinctDocumentMiddleware | MongooseQueryAndDocumentMiddleware;
 
+  type MongooseRawResultQueryMiddleware = 'findOneAndUpdate' | 'findOneAndReplace' | 'findOneAndDelete' | 'findByIdAndUpdate' | 'findOneAndDelete';
```

Comment:
`findByIdAndUpdate` doesn't have distinct middleware, that function fires `findOneAndUpdate` middleware

Reply:
Also it looks like you have `findOneAndDelete` twice here

Reply:
Thanks @vkarpov15,  I'll update.